src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2699,7 +2700,9 @@ private Variables GetOrCreateNestedFunctionVariables(Variables container, Method
             Debug.Assert(!useDelegateInvokeParameterTypes || delegateInvokeMethod is object);
             Debug.Assert(!useDelegateInvokeReturnType || delegateInvokeMethod is object);
 
-            var oldSymbol = this.CurrentSymbol;
+            var oldSymbol = this._symbol;
```

Comment:
:memo: `_symbol` drives various APIs, such as `MethodParameters`

Reply:
Did you consider modifying the relevant APIs to instead check `CurrentSymbol`?

Reply:
Took another look, but the comment on `_symbol` says its primary purpose is to provide method parameters and return type. So I think this is correct.